src/Controllers/ShareDraftController.php

```diff
@@ -91,6 +91,10 @@ public function preview(HTTPRequest $request)
         $page = Versioned::get_by_stage(SiteTree::class, Versioned::DRAFT)
             ->byID($shareToken->PageID);
 
+        if (!$page) {
```

Comment:
Please move the check after the extension point as it can update the `$page` variable.